crates/polars-core/src/hashing/vector_hasher.rs

```diff
@@ -22,11 +24,15 @@ pub trait VecHash {
     /// Compute the hash for all values in the array.
     ///
     /// This currently only works with the AHash RandomState hasher builder.
-    fn vec_hash(&self, _random_state: RandomState, _buf: &mut Vec<u64>) {
+    fn vec_hash(&self, _random_state: RandomState, _buf: &mut Vec<u64>) -> PolarsResult<()> {
         unimplemented!()
     }
 
-    fn vec_hash_combine(&self, _random_state: RandomState, _hashes: &mut [u64]) {
+    fn vec_hash_combine(
+        &self,
+        _random_state: RandomState,
+        _hashes: &mut [u64],
+    ) -> PolarsResult<()> {
```

Comment:
Now that we return an error can we use the `polars_err!(op == vec_hash)` macros to return a nice error here?

Reply:
It seems that `polars_err!(op = vec_hash)` require a `dtype` which we can't extracted from `VecHash` trait. I introduce a new branch in `polar_err` for un-implemented case. WDYT?

Reply:
Yeap, good solution. :+1: